daemon/daemon.go

```diff
@@ -870,7 +876,10 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error)
 
 	daemonRepo := path.Join(config.Root, "containers")
 
-	if err := os.MkdirAll(daemonRepo, 0700); err != nil && !os.IsExist(err) {
+	if err := os.MkdirAll(daemonRepo, 0711); err != nil && !os.IsExist(err) {
```

Comment:
Yes, for dockerinit to be able to mount the root file system as docker-root user.